Color picker + align/distribute popup menus

smbea opened the pull request:
Related to https://github.com/camunda/improved-canvas/issues/5

smbea commented:
I mentioned in the weekly we could possibly open to bottom of context pad if there is no space on top. While I like that idea, it also was mentioned that we should keep this behaviour for all popup menus the same. And they do indeed behave the same at the moment:

https://github.com/camunda/improved-canvas/assets/25825387/bd9923ec-3c35-4b40-b2b8-dc5b4fc12e40

We could look into not covering the context pad as a core improvement potentially. But it shouldn't block this. And in this case, it's rare the cases where the popup menu will be cut off anyway since it's quite small

barmac commented on lib/features/popupMenu/HorizontalPopupMenu.js:
```diff
@@ -0,0 +1,46 @@
+
+}
+
+HorizontalPopupMenu.prototype._addGroupSeparator = function(e) {
+  const popupMenu = domQuery('.djs-popup');
+
+  if (!popupMenu.classList.contains('color-picker')
+      && !popupMenu.classList.contains('align-elements')) {
+    return;
+  }
+
+  const groups = domQueryAll('.djs-popup-group', popupMenu);
+
+  forEach(groups, function(group) {
+
+    if (group === groups[0]) {
+      return;
+    }
+
+    group.insertAdjacentHTML('beforebegin', '<div class="separator"/>');
+  });
+};
```
In order to simplify the code, we could use CSS instead. I wrote this snippet for context-pad, and it could be adjusted to serve popup-menu as well:

```css
.bjs-container .djs-context-pad .group:not(:last-child):after {
    display: block;
    content: "";
    width: 1px;
    height: 18px;
    background: var(--context-pad-divider-color);
    margin: auto 4px;
}
```

barmac replied:
So there is no need for mutation observer, queries etc.

smbea replied:
Ah that's interesting indeed. I'll try it out

barmac commented:
The contrast between separator and the background is super low. Is it necessary to be there if it's so hard to notice?

barmac commented:
What I just noticed is that the selected popup menu button (paint brush) is a rectangular while the hovered popup menu buttons are round rectangulars.

smbea commented:
> The contrast between separator and the background is super low. Is it necessary to be there if it's so hard to notice?

I personally like that it's there but subtle. We can discuss this with @YanaSegal too.

> I noticed that the first item is selected per default, even though it's not the value that is in fact selected.

This is a color picker issue. You can also see it in https://demo.bpmn.io/ . I can open an issue and follow up

I will have a look at the other hover aspects you mentioned.

smbea commented:
I implemented your suggestions @barmac.

* I replaced the groups logic with css (for context pad, I added it to an already opened PR with context pad issues: https://github.com/camunda/improved-canvas/pull/10)

* The issue of the first element being selected is a popup menu feature (happens for normal popups, color picker and align). We do that intentionally and we treat "hover" as `.selected`. This is probably related to keyboard navigation purposes. But in these cases, it looks weird. I will treat this as a follow up to properly consider how to approach this.

barmac commented on assets/popup-menu.css:
```diff
@@ -0,0 +1,63 @@
+  fill: white;
+}
+
+.djs-popup.color-picker .separator,
+.djs-popup.align-elements .separator {
+  width: 1px;
+  height: 18px;
+  background: var(--context-pad-divider-color);
+  margin: 0 4px;
+}
```
To clean up:

```suggestion
```

smbea commented:
@barmac I updated with your last suggestions 🙂

barmac reviewed (approved):
I added on top:
- ES6-ification https://github.com/camunda/improved-canvas/pull/9/commits/777ad825d9b236e4a697a57a73fd0b2d8dfe6c92
- modern CSS https://github.com/camunda/improved-canvas/pull/9/commits/2994ebb4687253c395bb21309886639fa844bc05 (shaved some lines and allows to easier maintain the code)

Let's merge this! The context pad looks cool
